src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -87,24 +87,21 @@ private void LogRequestStarted(AuthenticationRequestParameters authenticationReq
             authenticationRequestParameters.RequestContext.Logger.InfoPii(messageWithPii, messageWithoutPii);
         }
 
-        protected SortedSet<string> GetDecoratedScope(SortedSet<string> inputScope)
+        protected virtual SortedSet<string> GetDecoratedScope(SortedSet<string> inputScope)
         {
-            SortedSet<string> set = new SortedSet<string>(inputScope.ToArray());
-            set.UnionWith(ScopeHelper.CreateSortedSetFromEnumerable(OAuth2Value.ReservedScopes));
+            // OAuth spec states that scopes are case sensitive, but 
+            // merge the reserved scopes in a case insensitive way, to 
+            // avoid sending things like "openid OpenId" (note that EVO is tollerant of this)
```

Comment:
nit: should you put "Microsoft identity platform" instead of EVO? more specific and other IdPs maybe case sensitive and following the spec